crates/storage/src/key.rs

```diff
@@ -574,4 +714,66 @@ mod tests {
+    ];
+
+    const DEC128_SHIFT: u128 = 10_u128.pow(18);
+
+    fn with_sign<const N: usize>(s: u8, b: [u8; N]) -> Vec<u8> {
+        let mut bytes = vec![s];
+        bytes.extend_from_slice(&b);
+        bytes
+    }
+
+    #[test_case(b"slice".as_slice(), b"slice"; "slice")]
+    #[test_case(b"Vec".to_vec(), b"Vec"; "vec_u8")]
+    #[test_case("str", b"str"; "str")]
+    #[test_case("String".to_string(), b"String"; "string")]
+    #[test_case(Addr::from_array(*b"ThisIsAValidAddress-"), b"ThisIsAValidAddress-"; "addr")]
+    #[test_case(&Addr::from_array(*b"ThisIsAValidAddress-"), b"ThisIsAValidAddress-"; "borrow_addr")]
+    #[test_case(Hash::<32>::from_array([1;32]), &[1; 32]; "hash")]
+    #[test_case(&Hash::<20>::from_array([2;20]), &[2; 20]; "borrow_hash")]
+    #[test_case(Duration::from_nanos(100), &100_u128.to_be_bytes(); "duration")]
+    #[test_case((10_u32, 265_u32 ), &DOUBLE_TUPLE_BYTES; "double_tuple")]
+    #[test_case(("Hello".to_string(), 10_u32, "World".to_string()), &DOUBLE_TRIPLE_BYTES; "triple")]
+    /// ---- Rust native numbers ----
+    #[test_case(10_u64, &10_u64.to_be_bytes(); "u64_10")]
+    #[test_case(-1_i8, &[127]; "i8_neg_1")]
+    #[test_case(1_i8, &[129]; "i8_1")]
+    /// ---- Unisgned integers ----
+    #[test_case(Uint64::new(10), &10_u64.to_be_bytes(); "uint64_10")]
+    #[test_case(Uint128::new(10), &Uint128::new(10).to_be_bytes(); "uint128_10")]
+    #[test_case(Uint256::MIN, &[0; 32]; "uint256_MIN")]
+    #[test_case(Uint512::MAX, &Uint512::MAX.to_be_bytes(); "uint256_MAX")]
+    /// ---- Unisgned Decimals ----
+    #[test_case(Udec128::from_str("10").unwrap(), &(10 * DEC128_SHIFT).to_be_bytes(); "udec128_10")]
+    #[test_case(Udec128::from_str("5.5").unwrap(), &(5 * DEC128_SHIFT + DEC128_SHIFT / 2).to_be_bytes(); "udec128_5.5")]
+    #[test_case(Udec128::MIN, &Uint128::MIN.to_be_bytes(); "udec128_0")]
+    #[test_case(Udec256::MAX, &Uint256::MAX.to_be_bytes(); "udec256_MAX")]
+    /// ---- Signed integers ----
+    #[test_case(Int64::new_positive(10_u64.into()), &with_sign(1, 10_u64.to_be_bytes()); "int64_10")]
+    #[test_case(Int128::new_negative(10_u64.into()), &with_sign(0, (u128::MAX - 10).to_be_bytes()); "int128_neg_10")]
+    #[test_case(Int256::MIN, &with_sign(0, Uint256::MIN.to_be_bytes()); "int256_MIN")]
+    #[test_case(Int256::MAX, &with_sign(1, Uint256::MAX.to_be_bytes()); "int256_MAX")]
+    /// ---- Signed Decimals ----
+    #[test_case(Dec128::MAX, &with_sign(1, Uint128::MAX.to_be_bytes()); "dec128_MAX")]
+    #[test_case(Dec128::MIN, &with_sign(0, Uint128::MIN.to_be_bytes()); "dec128_MIN")]
+    #[test_case(Dec256::from_str("-10.5").unwrap(), &with_sign(0, (Uint256::MAX - Uint256::from(10 * DEC128_SHIFT + DEC128_SHIFT / 2)).to_be_bytes()); "dec128_neg_10_5")]
+    #[test_case(Dec256::from_str("20.75").unwrap(), &with_sign(1, (Uint256::from(20 * DEC128_SHIFT + DEC128_SHIFT / 2 + DEC128_SHIFT / 4)).to_be_bytes()); "dec128_20_75")]
```

Comment:
Can you make this more readable by breaking the lines as in here: https://github.com/left-curve/grug/pull/131/commits/b40fc0a7b154f313f567606e8bab449d33f7b1b0#diff-df003dd0c2a4fd9d4d09c64190f2a81bc84fd5e4afd252c510c553ee68c44d40L112-R137